src/elements/content-sidebar/activity-feed/comment/BaseComment.js

```diff
@@ -157,23 +154,30 @@ const BaseComment = (props: BaseCommentProps) => {
         onSelect(false);
     };
 
-    const handleMessageUpdate = ({
-        id: messageID,
-        text,
-        hasMention,
-    }: {
-        hasMention: boolean,
-        id: string,
-        text: string,
-    }): void => {
-        onEdit(messageID, text, undefined, hasMention, permissions);
+    const handleMessageUpdate = ({ id: messageID, text, hasMention }) => {
+        if (onAnnotationEdit) {
+            onAnnotationEdit({ id: messageID, text, permissions });
+        } else if (onCommentEdit) {
+            onCommentEdit({
+                id: messageID,
+                text,
+                hasMention,
+                permissions,
+            });
+        }
         commentFormSubmitHandler();
     };
 
     const handleStatusUpdate = (selectedStatus: FeedItemStatus): void => {
-        onEdit(id, undefined, selectedStatus, false, permissions);
+        if (onAnnotationEdit) {
+            onAnnotationEdit({ id, permissions });
+        } else if (onCommentEdit) {
+            onCommentEdit({ id, status: selectedStatus, hasMention: false, permissions });
+        }
     };
 
+    const onEdit = onCommentEdit ?? onAnnotationEdit;
+
     const canDelete = permissions.can_delete;
     const canEdit = onEdit !== noop && permissions.can_edit;
     const canResolve = onEdit !== noop && permissions.can_resolve;
```

Comment:
I see this was already here but wondering if the noop check will cover the case of onCommentEdit and onAnnotationEdit both being null/undefined?

Reply:
Just got to where `onEdit={noop}` was deleted in Replies component and now neither of the edit props are passed in. I'm not familiar with how noop works though so maybe it will still work (because in your sceenshots/demo it seems to work). Just wanted to add this specific example to confirm.

Reply:
Ah, I see what you're seeing here now, functionality wise. Yes, editing replies does not work. I think we would need to pass `onComment` edit through to `Replies` even for annotations. (Maybe another good reason to pass replies in as a child instead).

Reply:
Eventually, I found that noop is a default coming all the way down from ActivitySidebar, so I reverted this use noop. Tragically, passing onCommentEdit to BaseComment in Replies does not enable modifying replies. But on the bright side, I learned that modifying replies is out of scope here. Thanks for bringing this up.